### PR TITLE
Google+/Twitter/GNU Social: Prevent posting loops when mirroring posts

### PR DESCRIPTION
**Pull request by annando**
Sometimes it happens that old posts are reposted through the mirror functionality. We now add some parameter ("message_id") that will be used as URI for these posts. Since the URI is unique on Friendica, there should be no double posts anymore.

This will currently only work for the "develop" branch of Friendica. But it will do no harm to the "master" branch.

Additionally there is some small programming that adds fetched keywords from links that are fetched from Google+.

### CODE REVIEW

File: fromgplus/fromgplus.php [hunk 1]
```diff
@@ -56,6 +58,9 @@ function fromgplus_addon_settings(&$a,&$s) {
 	$s .= '<label id="fromgplus-label" for="fromgplus-account">'.t('Google Account ID').' </label>';
 	$s .= '<input id="fromgplus-account" type="text" name="fromgplus-account" value="'.$account.'" />';
 	$s .= '</div><div class="clear"></div>';
+	$s .= '<label id="fromgplus-keywords-label" for="fromgplus-keywords">'.t('Add keywords to post').'</label>';
```
Comment (annando) [inline fromgplus/fromgplus.php]: I don't know how to do it.

Comment (tobiasd) [inline fromgplus/fromgplus.php]: With the `util/run_xgettext.sh` tool, which calls the xgettext program.

Change into the addon directory, say `addon/twitter` and call the tool with the `-s` parameter, e.g. `$>../../util/run_xgettext.sh -s` then the new messages.po file will be created in `addon/twitter/lang/C/messages.po` where Transifex can pic it up.

Comment (tobiasd) [inline fromgplus/fromgplus.php]: Yes, the `-a` parameter works as well. But I think when you are working on an addon and happen to update the strings, then the 1st call is more easy to remember.

Comment (annando) [inline fromgplus/fromgplus.php]: This week I'm busy since I'm working at a customer. I will try to do it in the next days.

**Pull request merged**